java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -860,18 +878,24 @@ class DatabaseImpl(
                     LEFT JOIN collection_entries
                         ON fields.is_collection = 1
                         AND collection_entries.collection_id = field_values.value_id
-                    WHERE fields.type_id = ?
+                    WHERE fields.type_id in (${typeIds.map { it.toString() }.joinToString()})
                 """.trimIndent()
 
             delete(
                 TABLE_NUMBER_PRIMITIVES,
-                "id NOT IN ($usedFieldIdsQuery)",
-                arrayOf(PrimitiveType.Number.ordinal.toString())
+                "id NOT IN (${usedFieldIdsQuery(
+                    listOf(
+                        PrimitiveType.Number.ordinal,
+                        PrimitiveType.Float.ordinal,
+                        PrimitiveType.Double.ordinal
```

Comment:
It would be good to share this with the numeric_types var above, eg add a const to the companion
private const val NUMERIC_TYPES =  listOf(
            PrimitiveType.Number.ordinal,
            PrimitiveType.Float.ordinal,
            PrimitiveType.Double.ordinal
        )
with a comment explaining these are the types stored in the TABLE_NUMBER_PRIMITIVES